### DISCUSSION TITLE
Whitelist good styles

### DISCUSSION THREAD

**Discussion by corysimmons** [shellscape/jsx-email#47] [Ideas]
I noticed

> Note: Most email clients are style-limited and some styles may not work.

Is there any way to whitelist only the cross-client-friendly styles and throw an error if someone tries to use bad ones?

I'm not sure how to know which ones work unless I have a dozen email clients open at the same time.

**Comment by shellscape** [comment]
Solid idea. We could bake in caniemail in some fashion. Would be cool if there was a threshold for % of support or some such

**Reply by corysimmons** [reply]
Nice, I didn't even know that list existed!

https://www.caniemail.com/api/data.json ‼️

**Reply by shellscape** [reply]
If you get the itch to use that to add to jsx-email, please do! ❤️ contributions

**Reply by shellscape** [reply]
couple of npm packages that could be used for inspiration https://www.npmjs.com/search?q=caniemail

**Reply by corysimmons** [reply]
I think I know how to approach it. Just trying to talk my boss into funding it :) and I'll put it in their org. I don't make enough newsletters personally for it to really save me a bunch of time but I can see it becoming big (kinda like the "Autoprefixer of HTML Emails").

**Comment by shellscape** [comment]
<img width="494" alt="Screenshot 2023-11-03 at 12 28 40 AM" src="https://github.com/shellscape/jsx-email/assets/76371/492cd4b8-c0f4-4083-b4e5-1dee47978d11">

I've got something cooking :)

**Reply by corysimmons** [reply]
This is beautiful. 👌 Quick work!

Just curious what CLI lib you reach for? I used to use Meow/Chalk back in the day but I'm not sure what's good now.

**Reply by corysimmons** [reply]
An ESLint plugin might get complicated. You'd have to do something like Autoprefixer and allow users to pick the clients or % they care about, or it'd be annoying having every tag with a red squiggle under it because some ancient client doesn't support anything.

**Reply by shellscape** [reply]
yeah I use meow when things get hairy. chalk is my goto. the jsx-email cli is just yargs-parser + chalk.
